src/utils/balance.ts

```
@@ -22,15 +22,26 @@ export const formatBalance = (value: bigint | string, decimals: number) => {
-  if (Math.abs(num) >= 1_000_000_000) {
-    return (num / 1_000_000_000).toFixed(2) + 'B';
-  } else if (Math.abs(num) >= 1_000_000) {
-    return (num / 1_000_000).toFixed(2) + 'M';
-  } else if (Math.abs(num) >= 1_000) {
-    return (num / 1_000).toFixed(2) + 'K';
-  } else {
-    return num.toFixed(2);
+export function formatReadable(num: number | string): string {
+  if (typeof num === 'string') {
+    const parsed = parseFloat(num);
+    if (isNaN(parsed)) return num;
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)